src/ledger/shred.zig

```diff
@@ -529,6 +529,8 @@ pub fn GenericShred(
 pub const ShredId = struct {
     slot: Slot,
     index: u32,
+    // Question: Why have shred_type as part of the ShredId?
```

Comment:
its possible to have a data and code shred of the same slot and the same index, which without shred_type, would not be a unique struct -- to make the struct unique, we need the shred_type